core/src/main/java/org/kiwix/kiwixmobile/core/settings/SliderPreference.java

```diff
@@ -171,9 +170,12 @@ private void setGrayState(View view) {
 
     TextView titleView = view.findViewById(android.R.id.title);
     TextView summaryTV = view.findViewById(android.R.id.summary);
-    if (!enabled) {
-      titleView.setTextColor(Color.GRAY);
-      summaryTV.setTextColor(Color.GRAY);
+    if (enabled) {
+      titleView.setEnabled(true);
+      summaryTV.setEnabled(true);
+    } else {
+      titleView.setEnabled(false);
+      summaryTV.setEnabled(false);
```

Comment:
Can you try removing these statements & check if it works?
Otherwise, it can be shortened to:
```java
titleView.setEnabled(enabled);
summaryTV.setEnabled(enabled);
```
;)

Reply:
> Can you try removing these statements & check if it works?
> Otherwise, it can be shortened to:
> 
> ```java
> titleView.setEnabled(enabled);
> summaryTV.setEnabled(enabled);
> ```
> 
> ;)

Yes i did try this as well.
It keeps the zoom level enabled irrespective of the Zoom controls toggle state.
If you insist I'll try again and get back to you! :)

Reply:
> Can you try removing these statements & check if it works?
> Otherwise, it can be shortened to:
> 
> ```java
> titleView.setEnabled(enabled);
> summaryTV.setEnabled(enabled);
> ```
> 
> ;)

Hey @abdulwd thee code you suggested is not working .
It keeps the zoom level enabled irrespective of the zoom controls toggle state.

Reply:
this cannot be logically true as the code
```
if(myBoolean){
setEnabled(true)
} else{
setEnabled(false)
}
```
is identical to
```
setEnabled(myBoolean)
```

Android studio should give you warning and even still I would say the compiler rewrites it